src/arithmetic/list_funcs/list_funcs.c

```diff
@@ -105,6 +108,96 @@ SIValue AR_TOLIST(SIValue *argv, int argc, void *private_data) {
 	return array;
 }
 
+/* Convert a list of values to a list of boolean values.
+   The conversion of each item in list is done using toBooleanOrNull */
+SIValue AR_TOBOOLEANLIST(SIValue *argv, int argc, void *private_data) {
+	ASSERT(argc == 1);
+	if(SI_TYPE(argv[0]) == T_NULL) {
+		return SI_NullVal();
+	}
+	ASSERT(SI_TYPE(argv[0]) == T_ARRAY);
+	SIValue originalArray = argv[0];
+
+	// get array length
+	uint32_t arrayLen = SIArray_Length(originalArray);
+
+	SIValue array = SI_Array(arrayLen);
+	for(uint i = 0; i < arrayLen; i++) {
+		SIValue v = SIArray_Get(originalArray, i);
+		SIArray_Append(&array, AR_TO_BOOLEAN(&v, 1, NULL));
+		SIValue_Free(v);
```

Comment:
We might want to consider allowing direct access to SI_Array internal array, this way we won't have to go through `SIArray_Get` and call `SIValue_Free` on each element,
@AviAvni what do you think?